### PR TITLE
Detach validation api

### PR DESCRIPTION
**Pull request by eliranmal**
enable using rendering and validation separately

- expose `validate()` as public api
- allow bypassing validation in render options (validates by default)

### CODE REVIEW

File: js/LPJsonPollock.js [hunk 1]
```diff
@@ -117,6 +110,21 @@ export default class LPJsonPollock {
     return frag;
   }
 
+	validate() {
```
Comment (meirotstein) [inline js/LPJsonPollock.js]: you missed the json param: `validate(json: Object|string)`

File: js/LPJsonPollock.js [hunk 2]
```diff
@@ -94,18 +94,11 @@ export default class LPJsonPollock {
...
-      // This will throws an error if fails
-      jsonObj = JSON.parse((json: any));
-    } else {
-      jsonObj = (json: any);
-    }
-    this.jsonValidator(jsonObj);
-    if (this.jsonValidator.errors) {
-      throw new JsonPollockError('Schema validation error, see \'errors\' for more details', this.jsonValidator.errors);
-    }
+	  if (!options.skipValidation) {
```
Comment (meirotstein) [inline js/LPJsonPollock.js]: Now the validation function is also parse the json object if it was given as a string, so if the `skipValidation` option is enabled we missed that part

Comment (eliranmal) [reply js/LPJsonPollock.js]: true! i fix!

Comment (eliranmal) [reply js/LPJsonPollock.js]: done!

File: js/LPJsonPollock.js (hunk 1 above)
Comment (eliranmal) [reply js/LPJsonPollock.js]: true that, fixed!

File: index.js [hunk 3]
```diff
@@ -9,6 +9,7 @@ const instance = new LPJsonPollock();
 
 const init = instance.init.bind(instance);
 const render = instance.render.bind(instance);
+const validate = instance.render.bind(instance);
```
Comment (eliranmal) [inline index.js]: is it an overlook?

change to `instance.validation.bind(instance)`

Comment (meirotstein) [reply index.js]: Right!

**Comment by meirotstein** [comment]
@eliranmal don't think this relevant anymore - I'm closing it for now. let's reopen if necessary.

**Pull request closed without merging**